mlir/lib/Dialect/EmitC/IR/EmitC.cpp

```diff
@@ -50,6 +50,26 @@ void mlir::emitc::buildTerminatedBody(OpBuilder &builder, Location loc) {
   builder.create<emitc::YieldOp>(loc);
 }
 
+/// Check that the type of the initial value is compatible with the operations
+/// result type.
```

Comment:
Passed separately, expectedType isn't necessarily op's return type. Since op is being passed in anyway for error emission, can instead drop expectedType and call op->getType() inside (also making more meaningful original error message below accurate).